Source/WebCore/dom/Element.cpp

```diff
@@ -4740,6 +4742,52 @@ bool Element::isSpellCheckingEnabled() const
     return true;
 }
 
+bool Element::isWritingSuggestionsEnabled() const
+{
+    // If none of the following conditions are true, then return `false`.
+
+    // `element` is an `input` element whose `type` attribute is in either the
+    // `Text`, `Search`, `URL`, `Email` state and is `mutable`.
```

Comment:
You're not testing for mutable here and below: negation of `isDisabledFormControl()`.